src/ophyd_async/core/detector.py

```diff
@@ -299,28 +299,26 @@ async def _prepare(self, value: T) -> None:
             exposure=self._trigger_info.livetime,
         )
 
-    @AsyncStatus.wrap
-    async def kickoff(self) -> None:
-        self._fly_status = AsyncStatus(self._fly(), self._watchers)
+    def kickoff(self, timeout: Optional[float] = None):
```

Comment:
The original code kicked off `self._observe_writer_indices` and returned immediately, then returned this status in `complete`. This code now actually awaits this Status, which is wrong.

However, I think a simplification of the code is the better fix:

```python
    @AsyncStatus.wrap
    async def kickoff(self) -> None:
        # Nothing to do in kickoff, we already armed in prepare
        pass

    @WatchableAsyncStatus.wrap
    async def complete(self):
        async for index in self.writer.observe_indices_written(self._frame_writing_timeout):    
            yield WatcherUpdate(
                name=self.name,
                current=index,
                initial=self._initial_frame,
                target=end_observation,
                unit="",
                precision=0,
                time_elapsed=time.monotonic() - self._fly_start,
            )
            if index >= end_observation:
                break
```
Then can delete `_fly` and `_observe_writer_indices`...

Reply:
`Flyable` tells us that `kickoff` and `complete` must both return statuses.

Reply:
That's fine, they're both being wrapped by `*AsyncStatus` so return a `Status`. It's just that `kickoff` returns one that is immediately done...